feat: Add mandatory license field to job submission form

somendrajaat opened the pull request:
Adds a required field for project licenses to ensure all job postings are from properly licensed open source projects. The field includes:
- Mandatory textarea for license documentation
- Help text explaining the requirement
- Support for various license documentation formats (repo links, webpages, etc.)

This change helps maintain OSD's commitment to open source principles by ensuring all job postings are from properly licensed projects.

somendrajaat commented:
Hi @Erioldoesdesign , thank you for the warm welcome!

The HTML file is just the form template that users fill out. When someone submits the form, the data is sent to Staticman, which then creates a new Markdown (.md) file for each job post. So, the HTML form populates the .md files—not the other way around. The HTML file itself doesn’t get updated by the Markdown files.
If we add a new field to the form (like the license field), that information will show up in new .md files generated from future submissions.

SaptakS commented on job-form.html:
```diff
@@ -44,6 +44,16 @@ <h1>Submit a Job</h1>
+		<div class="form-group">
+		  <label for="license">Project License/s *</label>
+		  <p class="help-block">Project must have some kind of open source license publicly viewable to be accepted. You can share any kind of document e.g. link to the license in your public repo, a webpage, an online cloud document etc. We want to be sure that you are following open source principles.</p>
+		  <textarea class="form-control"
```
Why a textarea and not an `input type="url"`? I think mostly it will be a link? Thoughts @Erioldoesdesign ?

somendrajaat replied:
@SaptakS  done, thanks for suggestion!

SaptakS reviewed (changes requested):
@somendrajaat Thank you for opening this PR! I haven't tested it fully but left some comments asking some minor changes. Also, I think we should also update the [markdown template file](https://github.com/opensourcedesign/jobs/blob/master/job-template.md?plain=1)

You will also need to update [the list](https://github.com/opensourcedesign/jobs/blob/master/staticman.yml#L10-L13) in the staticman config

somendrajaat commented:
Thank you for the feedback and suggestions!
I’ve reverted the curly quotes in the explanatory text to keep the original style.
I’ve added the new license field to the Staticman config so that form submissions with this field will be accepted.
I’ve also updated the Markdown job template to include the new license field, with a comment to guide contributors.
Let me know if there’s anything else I should address.
